### DISCUSSION TITLE
SSD1306 stop working when starting ADS1115 on ESP32

### DISCUSSION THREAD

**Discussion by Zzo911** [olikraus/u8g2#1790] [General]
Dear all,

I have setup a project on a ESP32 and a SSD1306 LCD to display some logs instead of using the Serial.
Everything run fine.

````
#include <U8g2lib.h>
U8G2_SSD1306_128X64_NONAME_1_SW_I2C LCD(U8G2_R0, /* clock=*/ SCL, /* data=*/ SDA, /* reset=*/ U8X8_PIN_NONE);

#define LCDLOG_WIDTH  25
#define LCDLOG_HEIGHT  6

uint8_t LCDlog_buffer[LCDLOG_WIDTH*LCDLOG_HEIGHT];
LCDlog.begin(LCDLOG_WIDTH, LCDLOG_HEIGHT, LCDlog_buffer);
LCDlog.setLineHeightOffset(0);
LCDlog.setRedrawMode(0);
````

Then I have a function adding lines to LCDlog, working perfect.
So I have added a second device on the I2C bus which is an ADS1115 to monitor a battery.
This module works fine also on its own.

````
#include <ADS1X15.h>

ADS1115 ADS(0x48);
ADS.begin();
````

But as soon as the ADS.begin() starts after the u8g2 part, the SSD1306 freeze and do not update anymore?
I ran a little ino code that check the devices connect to the I2C bus and it detects both devices, so it looks OK.
Running the same code including u8g2 and ADS on an Arduino MEGA works but not an ESP32.
I have tested several ADS libraries and the LCD freeze at the same place.

When using the Adafruit_GFX.h and the ADS1X15.h  on ESP32 everything works, but I need to use u8g2 as I prefer.
Is there some restriction with the u8g2 running on ESP32 and another devices on the I2C bus?

Many thanks

**Comment by v1993** [comment]
Use HW constructor if you want to have few different devices on same pins.

**Reply by Zzo911** [reply]
Oh, many thanks for that I use the following constructor and it's working now.
U8G2_SSD1306_128X64_NONAME_1_HW_I2C LCD(U8G2_R0, /* reset=*/ U8X8_PIN_NONE);
